### DISCUSSION TITLE
Can php workers be on another server?

### DISCUSSION THREAD

**Discussion by lezhnev74** [roadrunner-server/roadrunner#1037] [Q&A] [answered]
Hey! I was playing with the RR, I am new to this software. So far I understood that both RR and PHP must be on the same machine, so RR can manage the pool of workers (which are called via `php script.php`).

My question is, can I have RR on a different machine (different docker service) and somehow let it manage workers on another machine? 

My reasonings:
1. It should be possible with PHP-FPM that accepts connections over the network. In this case RR does not manage the pool really and only proxies requests to fpm.
2. Supervisor can manage php workers on a separate machine. But then it is not clear how to establish a connection b/w RR and multiple independent php processes on a separate server.

Any hints will be greatly appreciated!

**Comment by rustatian** [comment, marked as answer]
Hey @lezhnev74. Thanks for the question. 
No, that's not possible.
Actually, you can have 2 RR instances. 1 acts like a proxy, the second - handles proxied requests.

**Reply by lezhnev74** [reply]
Aha, yeap, I understand. One instance can be configured to send requests to another instance, I will check the docs to see how such proxying can be achieved.
